### ISSUE TITLE
"X is defined but never used (no-unused-vars)" error when rules says to ignore the no-unused vars. Error only happen sometimes

### ISSUE THREAD

**Issue by nicholasfc** [eslint/eslint#12530]
**Tell us about your environment**

* **ESLint Version:** "babel-eslint": "^10.0.3" "eslint": "^5.16.0"
* **Node Version:** v10.16.3
* **npm Version:** 6.11.3

**What parser (default, Babel-ESLint, etc.) are you using?** Babel-ESLint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
"eslintConfig": {
    "root": true,
    "env": {
      "node": true
    },
    "extends": [
      "plugin:vue/essential",
      "eslint:recommended"
    ],
    "rules": {
      "no-console": "off",
      "no-unused-vars": [
        "error",
        {
          "vars": "all",
          "args": "after-used",
          "ignoreRestSiblings": false
        }
      ]
    }
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**
I used the npm run serve for a vue.js project

```js
methods: {
    onSubmit() {
      db.collection("members")
        .add({
          **
        })
        .then(docRef => {
          this.$router.push("/");
        })
        .catch(err => console.log(err));
    }
  }
```

```bash
npm run serve
```

**What did you expect to happen?**
I expected it to compile and run the service and have the website online but half of the time the linter throws the no-unused var error so I have to run the npm run serve command a bunch of times for it to finally compile and work, even though I have a rule to ignore the no-unused vars

**What actually happened? Please include the actual, raw output from ESLint.**
ERROR  Failed to compile with 1 errors                            3:13:56 PM

 error  in ./src/views/AddPlayer.vue

Module Error (from ./node_modules/eslint-loader/index.js):
error: 'docRef' is defined but never used (no-unused-vars) at src/views/AddPlayer.vue:49:15:
  47 |           comments: this.comments
  48 |         })
> 49 |         .then(docRef => {
     |               ^
  50 |           this.$router.push("/");
  51 |         })
  52 |         .catch(err => console.log(err));

**Are you willing to submit a pull request to fix this bug? No, not enough knowledge to do so**

**Comment by kaicataldo** [comment]
A few questions:

1. Do you also see this behavior when running ESLint directly from the command line (if not, this sounds like an issue with your compiler setup)?
1. Can you confirm this also happens with the latest version of ESLint and babel-eslint?
1. Given the snippet of code you shared, it looks like it _should_ be erroring, since `docRef` isn't being used.

**Comment by nicholasfc** [comment]
I'm sorry but I already changed my rules to "no-unused-vars": "off" and it seems to have "fixed" since I don't get the error anymore. 

Maybe the problem was the rules I was using and as you said, the linter is working as supposed.
